test/galore/README.md

```diff
@@ -0,0 +1,117 @@
+  - `adamw*.{json,html} - reference `torch` `AdamW`
+  - `galore_adamW-128*.{json,html}` - reference `GaLoreAdamW` optimizer from the original `GaLore` repo with `rank=128`
+  - `galore_adamW-64*.{json,html}` - reference `GaLoreAdamW` optimizer from the original `GaLore` repo with `rank=64`
+
+The provided notebook has the plots and stats of the memory traces of these sample runs embedded and a cursory analysis of these sample data as follows:
+
+- CUDA Memory Timeline
+
+  - Plots of the html memory timeline from `torch.profiler`
+
+    - Reference `torch.optim.AdamW`
+
+      ![alt text](./profiler_out/adamw_timeline.png)
+
+    - Reference `GaLoreAdamW`
+
+      ![alt text](./profiler_out/galore_adamW-128_timeline.png)
+
+We can see a long delay between the first backwards step for `GaLoreAdamW` due to the calculation of the projection matrix (calls `torch.linalg.svd` on the `grad`).
+
+- Memory Stats
+
+  - Summary stats for the memory timeline are generated by parsing the `json` output of the torch profiler [memory trace](https://pytorch.org/docs/main/profiler.html#torch.profiler._KinetoProfile.export_memory_timeline).
+
+    - Memory Usage (in MB) for `torch.optim.AdamW`
+
+    |        | Parameter | Optimizer_State | Input | Temporary | Activation | Gradient | Autograd_Detail | Unknown |
+    | ------ | --------- | --------------- | ----- | --------- | ---------- | -------- | --------------- | ------- |
+    | mean   | 381.9     | 628.1           | 0.0   | 0.2       | 356.0      | 162.8    | 6.6             | 29.5    |
+    | min    | 381.9     | 0.0             | 0.0   | 0.0       | 0.0        | 0.0      | 0.0             | 0.0     |
+    | median | 381.9     | 769.6           | 0.0   | 0.0       | 337.8      | 171.7    | 3.1             | 16.3    |
+    | max    | 382.0     | 769.6           | 0.3   | 6.7       | 1,338.1    | 395.7    | 312.9           | 402.8   |
+
+    - Memory Usage (in MB) for `GaLoreAdamW` reference, rank 128
+
+    |        | Parameter | Optimizer_State | Input | Temporary | Activation | Gradient | Autograd_Detail | Unknown |
+    | ------ | --------- | --------------- | ----- | --------- | ---------- | -------- | --------------- | ------- |
+    | mean   | 491.4     | 349.0           | 0.0   | 0.2       | 226.6      | 246.1    | 4.2             | 18.7    |
+    | min    | 381.9     | 0.0             | 0.0   | 0.0       | 0.0        | 0.0      | 0.0             | 0.0     |
+    | median | 516.3     | 403.6           | 0.0   | 0.0       | 75.7       | 272.8    | 0.0             | 18.1    |
+    | max    | 595.0     | 403.6           | 0.3   | 6.6       | 1,336.0    | 395.3    | 312.9           | 173.6   |
+
+  - The `optimizer state` is indeed smaller for the `GaLoreAdamW` optimizer. Interestingly, the `Parameter` sizes balloons in the `GaLore` optimizer, likely due to extra data copies. Admittedly, the implementation is only a reference (per original repo) and leaves much room for optimization.
```

Comment:
thank you for noting this, how are you thinking about next steps here? One hope I had for merging this is for other repos to take a dependency like https://github.com/pytorch/torchtune